handlers/rocketchat/rocketchat.go

```diff
@@ -0,0 +1,142 @@
+package rocketchat
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/buger/jsonparser"
+	"github.com/nyaruka/courier"
+	"github.com/nyaruka/courier/handlers"
+	"github.com/nyaruka/courier/utils"
+	"github.com/nyaruka/gocommon/urns"
+	"net/http"
+)
+
+const (
+	configBaseURL     = "base_url"
+	configSecret      = "secret"
+	configBotUsername = "bot_username"
+)
+
+func init() {
+	courier.RegisterHandler(newHandler())
+}
+
+type handler struct {
+	handlers.BaseHandler
+}
+
+func newHandler() courier.ChannelHandler {
+	return &handler{handlers.NewBaseHandler(courier.ChannelType("RC"), "RocketChat")}
+}
+
+// Initialize is called by the engine once everything is loaded
+func (h *handler) Initialize(s courier.Server) error {
+	h.SetServer(s)
+	s.AddHandlerRoute(h, http.MethodPost, "receive", h.receiveMessage)
+	return nil
+}
+
+type Attachment struct {
+	Type string `json:"type"`
+	URL  string `json:"url"`
+}
+
+type moPayload struct {
+	User struct {
+		URN      string `json:"urn"     validate:"required"`
+		Username string `json:"username"`
+		FullName string `json:"full_name"`
+	} `json:"user" validate:"required"`
+	Text        string       `json:"text"`
+	Attachments []Attachment `json:"attachments"`
+}
+
+// receiveMessage is our HTTP handler function for incoming messages
+func (h *handler) receiveMessage(ctx context.Context, channel courier.Channel, w http.ResponseWriter, r *http.Request) ([]courier.Event, error) {
+	// check authorization
+	secret := channel.StringConfigForKey(configSecret, "")
+	if fmt.Sprintf("Token %s", secret) != r.Header.Get("Authorization") {
+		return nil, courier.WriteAndLogUnauthorized(ctx, w, r, channel, fmt.Errorf("invalid Authorization header"))
+	}
+
+	payload := &moPayload{}
+	err := handlers.DecodeAndValidateJSON(payload, r)
+	if err != nil {
+		return nil, handlers.WriteAndLogRequestError(ctx, h, channel, w, r, err)
+	}
+
+	// check content empty
+	if payload.Text == "" && len(payload.Attachments) == 0 {
+		return nil, handlers.WriteAndLogRequestError(ctx, h, channel, w, r, errors.New("no text or attachment"))
+	}
+
+	urn, err := urns.NewURNFromParts(urns.RocketChatScheme, payload.User.URN, "", payload.User.Username)
```

Comment:
@koallann latest gocommon is now in master if you want to update this PR

Reply:
I meant courier master

Reply:
Sorry, conflicts fixed 👍